web/index.html

```diff
@@ -461,71 +526,52 @@ <h3>Used in</h3>
-                    let usedElements = new Set();
-                    let result = [];
-                    craftingList.push(this.currentElement);
-
-                    while (craftingList.length > 0) {
-                        let thisElement = craftingList.shift();
-                        if (usedElements.has(thisElement)) {
-                            continue;
-                        }
-                        usedElements.add(thisElement);
-
-                        if (Object.keys(this.data.craftingCache).length == 0 || !this.data.craftingCache[thisElement]) {
-                            return []
-                        }
-                        let [left, right] = this.data.craftingCache[thisElement]
-
-                        result.unshift({ A: left, B: right, C: thisElement });
-
-                        if (this.data.costs[left] != 1) {
-                            craftingList.push(left);
-                        }
-
-                        if (this.data.costs[right] != 1) {
-                            craftingList.push(right);
-                        }
-                    }
-                    return result
-                },
                 elementCount: function () {
                     return Object.keys(this.data.index).length - 1;
                 },
                 isDefaultElement: function () {
                     return this.data.costs[this.currentElement] == 1;
-                }
+                },
+                isMobile: function () {
+                    return (/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent));
+                },
             },
             methods: {
+                getChunk: function (value) {
+                    let base64 = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-='
+                    let result = 0
+                    for (let i = 0; i < value.length; i++) {
+                        result = (result * 64 + base64.indexOf(value[i]))
+                    }
+                    return Math.floor(result / 100)
+                },
                 navigateTo: function (element) {
                     if (element != "") {
-                        if (!this.data.data[element]) {
-                            this.loadChunk(getChunk(element));
-                        }
-                        setQuery({ e: element });
+                        this.craftingTreeData = null;
+                        this.showHiddenRecipes = false;
                         if (this.currentElement == "") {
                             this.currentPanelIndex = 0;
+                            this.showFullRecipe = false;
+                        }
+                        if (!this.data.data[element]) {
+                            this.elementLoaded = false;
+                            this.craftingTreeData == null;
+                            this.loadChunk(this.getChunk(element)).then(_ => {
```

Comment:
Added a then statement following the chunk loading if the chunk has not already been loaded, which tells the components that the data is loaded and they can render, and initiates the scroll and crafting tree calculates